lib/rust/code-builder/src/lib.rs

```diff
@@ -56,13 +56,17 @@ impl CodeBuilder {
     /// Create a new line and insert appropriate indentation.
     pub fn newline(&mut self) {
         let space_count = self.spaces_in_indent * self.indent;
-        self.write(format!("\n{}", " ".repeat(space_count)));
+        self.buffer.reserve(space_count + 1);
+        self.buffer.push('\n');
+        for _ in 0..space_count {
+            self.buffer.push(' ');
+        }
         self.spaced = true;
     }
 
     /// Inserts a terminator symbol `;`.
     pub fn terminator(&mut self) {
-        self.write(";");
+        self.buffer.push(';');
```

Comment:
Why this change? Previous code was using an abstraction `self.write` to update the buffer. This is using internal value. It's longer, more complicated, and less modular. If you want to push a single char, create a function for it, next to `self.write`, or implement a trait which will consume either str or char.

Reply:
I wanted to avoid the unwraps here. Write/Format traits always return `Result`, but for strings this operation never errors. I'm fine with reducing the number of changes here if you don't like that.